config.go

```diff
@@ -880,6 +891,66 @@ func loadConfig() (*config, []string, error) {
 	return &cfg, remainingArgs, nil
 }
 
+// createDefaultConfig copies the file sample-btcd.conf to the given destination path,
+// and populates it with some randomly generated RPC username and password.
+func createDefaultConfigFile(destinationPath string) error {
+	// Create the destination directory if it does not exists
+	os.MkdirAll(filepath.Dir(destinationPath), 0700)
+
+	// We get the sample config file path, which is in the same directory as this file.
+	_, path, _, _ := runtime.Caller(0)
+	sampleConfigPath := filepath.Join(filepath.Dir(path), "sample-btcd.conf")
+
+	// We generate a random user and password
+	randomBytes := make([]byte, 20)
+	_, err := rand.Read(randomBytes)
+	if err != nil {
+		return err
+	}
+	generatedRPCUser := base64.StdEncoding.EncodeToString(randomBytes)
+
+	_, err = rand.Read(randomBytes)
+	if err != nil {
+		return err
+	}
+	generatedRPCPass := base64.StdEncoding.EncodeToString(randomBytes)
+
+	src, err := os.Open(sampleConfigPath)
+	if err != nil {
+		return err
+	}
+	defer src.Close()
+
+	dest, err := os.OpenFile(destinationPath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0700)
+	if err != nil {
+		return err
+	}
+	defer dest.Close()
+
+	// We copy every line from the sample config file to the destination,
+	// only replacing the two lines for rpcuser and rpcpass
+	reader := bufio.NewReader(src)
+	for err != io.EOF {
+		var line string
+		line, err = reader.ReadString('\n')
+		if err != nil && err != io.EOF {
+			return err
+		}
+
+		if strings.Contains(line, "rpcuser=") {
+			line = "  rpcuser=" + string(generatedRPCUser) + "\n"
```

Comment:
I don't believe this should have the extra 2 spaces at the front (nor the `rpcpass` below).

It ends up like the following:

```
; full set of credentials - limited or admin - or the RPC server will
; be disabled.
  rpcuser=<user>
  rpcpass=<pass>
; rpclimituser=whatever_limited_username_you_want
; rpclimitpass=
```

Instead of:

```
; full set of credentials - limited or admin - or the RPC server will
; be disabled.
rpcuser=<user>
rpcpass=<pass>
; rpclimituser=whatever_limited_username_you_want
; rpclimitpass=
```